### PR TITLE
[simple-oauth2] Allow http options for getToken

### PR DESCRIPTION
**Pull request by bertyhell**
According to the docs of simple-oauth2 the getToken function optionally accepts an httpOptions object and a scope property on the token config object
https://www.npmjs.com/package/simple-oauth2#authorization-code-flow

```javascript
// Get the access token object (the authorization code is given from the previous step).
const tokenConfig = {
  code: '<code>',
  redirect_uri: 'http://localhost:3000/callback',


  //-------------------- this -------------------------
  scope: '<scope>', // also can be an array of multiple scopes, ex. ['<scope1>, '<scope2>', '...']


};
 
// Optional per-call http options
const httpOptions = {};
 
// Save the access token
try {


  //-------------------- this -------------------------
  const result = await oauth2.authorizationCode.getToken(tokenConfig, httpOptions);


  const accessToken = oauth2.accessToken.create(result);
} catch (error) {
  console.log('Access Token Error', error.message);
}
```

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/simple-oauth2/index.d.ts [hunk 1]
```diff
@@ -113,7 +115,7 @@ export interface OAuthClient<ClientIdName extends string = 'client_id'> {
         ): string,
 
         /** Returns the Access Token object */
-        getToken(params: AuthorizationTokenConfig): Promise<Token>;
+        getToken(params: AuthorizationTokenConfig, httpOptions?: any): Promise<Token>;
```
Comment (netux) [inline types/simple-oauth2/index.d.ts]: According to [this commit on simple-oauth2](https://github.com/lelylan/simple-oauth2/commit/beb0c316b4fc8feab835071d77646daaf95bc641), `ownerPassword`, `clientCredentials` and `accessToken` also got this parameter added.

Comment (netux) [reply types/simple-oauth2/index.d.ts]: Also, what types can `httpOptions` be?

**Review by netux — CHANGES_REQUESTED** [review]
Pretty good, thanks! But some things have to be changed.

File: types/simple-oauth2/index.d.ts (hunk 1 above)
Comment (bertyhell) [reply types/simple-oauth2/index.d.ts]: Ok, added the param to the other functions

The type seems to be an http options object from the Wreck library:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/wreck/index.d.ts#L17-L41

I've copied the interface over

**Review by netux — APPROVED** [review]

**Pull request merged**